packages/react-server/src/ReactServerStreamConfigBrowser.js

```diff
@@ -21,24 +21,64 @@ export function flushBuffered(destination: Destination) {
   // transform streams. https://github.com/whatwg/streams/issues/960
 }
 
+let currentView = null;
+let writtenBytes = 0;
+
 export function beginWriting(destination: Destination) {}
 
 export function writeChunk(
   destination: Destination,
   chunk: PrecomputedChunk | Chunk,
 ): void {
-  destination.enqueue(chunk);
+  if (currentView === null) {
+    currentView = new Uint8Array(512);
+    writtenBytes = 0;
+  }
+
+  if (chunk.length > currentView.length) {
+    // this chunk is larger than our view which implies it was not
+    // one that is cached by the streaming renderer. We will enqueu
+    // it directly and expect it is not re-used
+    if (writtenBytes > 0) {
+      destination.enqueue(new Uint8Array(currentView.buffer, 0, writtenBytes));
+      currentView = null;
+      writtenBytes = 0;
+    }
+    destination.enqueue(chunk);
+    return;
+  }
+
+  const allowableBytes = currentView.length - writtenBytes;
+  if (allowableBytes < chunk.length) {
```

Comment:
`allowableBytes` could be zero if the previous write ended up filling the whole view. You could special case that to just enqueue the whole view.

Reply:
I’m curious about the performance of subarrays given they are fixed address ranges with no copying. Is there still significant overhead to subarray(0) that is worth avoiding with special cases?